geonode_mapstore_client/client/js/components/FeaturedList/FeaturedList.jsx

```diff
@@ -6,105 +6,14 @@
-import ResourceCard from '@js/components/ResourceCard';
 import FaIcon from '@js/components/FaIcon';
 import { withResizeDetector } from 'react-resize-detector';
-import { getResourceStatuses } from '@js/utils/ResourceUtils';
 import { actionButtons } from '@js/utils/ResourceServiceUtils';
-
-const Cards = withResizeDetector(({
-    resources,
-    formatHref,
-    isCardActive,
-    buildHrefByTemplate,
-    containerWidth,
-    width: detectedWidth,
-    options,
-    onResize,
-    actions,
-    onAction
-}) => {
-
-    const width = detectedWidth || containerWidth;
-    const margin = 24;
-    const size = 320;
-    const count = Math.floor(width / (size + margin));
-    const cardWidth = width >= size + margin * 2
-        ? Math.floor((width - margin * count) / count)
-        : '100%';
-    useEffect(() => {
-        onResize(count);
-    }, [ count ]);
-    const ulPadding = Math.floor(margin / 2);
-    const isSingleCard = count === 0 || count === 1;
-
-    const gridLayoutSpace = (idx) => {
-
-        const gridSpace = isSingleCard
-            ? {
-                width: width - margin,
-                margin: ulPadding
-            }
-            : {
-                width: cardWidth,
-                marginRight: (idx + 1) % count === 0 ? 0 : margin,
-                marginTop: 8
-            };
-
-        return gridSpace;
-    };
-
-    const containerStyle = isSingleCard
-        ? {
-            paddingBottom: 0
-        }
-        : {
-            paddingLeft: ulPadding,
-            paddingBottom: 0
-        };
-    return (
-        <ul
-            style={containerStyle}
-        >
-            {resources.map((resource, idx) => {
-                const {
-                    isProcessing,
-                    isDeleted
-                } = getResourceStatuses(resource);
-
-                if (isDeleted) {
-                    return null;
-                }
-
-                return (
-                    <li
-                        key={resource?.pk}
-                        style={(gridLayoutSpace(idx))}
-                    >
-                        <ResourceCard
-                            active={isCardActive(resource)}
-                            className={`${isDeleted ? 'deleted' : ''}`}
-                            data={resource}
-                            formatHref={formatHref}
-                            options={options}
-                            buildHrefByTemplate={buildHrefByTemplate}
-                            layoutCardsStyle="grid"
-                            loading={isProcessing}
-                            readOnly={isDeleted || isProcessing}
-                            featured
-                            actions={actions}
-                            onAction={onAction}
-                        />
-                    </li>
-                );
-            })}
-        </ul>
-    );
-});
+import Cards from '@js/components/Card';
```

Comment:
@DavidQuartz I don't understand how this fixes the bug, I only see the component moved outside. Could you please explain the fix?

Please move also the Cards component in `FeaturedList/Cards.jsx` because is currently used only here and it does not seems a generic component. Also keep the name of file and class consistent now there is Card and Cards

Reply:
@allyoucanmap The Cards component's extraction into its own component did not fix the bug, it was only done to keep up with the convention in React apps architecture. I will put it back if that's not ok.

Inside the Cards component, we previously used the following:
```javascript
    const count = Math.floor(width / (size + margin));
    const cardWidth = width >= size + margin * 2
```

It was replaced with the following (to limit the max number of resources requested to 4 per page):
```javascript
    const countNum = Math.floor(width / (size + margin));
    const count = countNum > 4 ? 4 : countNum; // limit count in order not to request for more than 4 per page
    const cardWidth = width >= size + margin * 2
```
This is because the count is wrong on the first render since it is calculated with the wrong width, and the featured resources are fetched and stored by the parent based on the wrong page size. In short, the wrong number of resources get supplied to the Cards component and it is forced to render them.

After the first fix, a check for `detectedWidth` is added to make sure the component only renders when its own detected width is available, and not its parent's width. Without this check, the bug is only half fixed.

These 2 fixes make sure the Cards component renders the right number of cards always.